### DISCUSSION TITLE
`flux bootstrap` but requires PR first

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#1804] [Q&A] [answered]
Is it possible to run `flux bootstrap`, but not committing it immediately to the main branch? We would like to have a code review process before merging it. It's a bit similar situation to push-branch for image-controller.

In an upgrade situation, this becomes more important, so we can double-check if any kustomize-patch we did will still work.
There is also a compliance process, where we have to ensure every commits to the main branch is reviewed by others.

**Comment by stefanprodan** [comment, marked as answer]
Here is a GitHub Action that opens a PR every time there is a Flux upgrade need https://github.com/fluxcd/flux2/tree/main/action#automate-flux-updates
